tests/multi_wlan/01_ap_sta.py

```diff
@@ -0,0 +1,117 @@
+# Basic Wi-Fi MAC layer test where one device creates an Access Point and the
+# other device connects to it as a Station. Also tests channel assignment (where
+# possible) and disconnection.
+
+try:
+    import network
+
+    network.WLAN
+except (ImportError, NameError):
+    print("SKIP")
+    raise SystemExit
+
+import os
+import sys
+import time
+
+CHANNEL = 8
+
+# Note that on slower Wi-Fi stacks this bumps up against the run-multitests.py
+# timeout which expects <10s between lines of output. We work around this by
+# logging something half way through the wait_for loop...
+CONNECT_TIMEOUT = 15000
+
+
+def wait_for(test_func):
+    has_printed = False
+    start = time.ticks_ms()
+    while not test_func():
+        time.sleep(0.1)
+        delta = time.ticks_diff(time.ticks_ms(), start)
+        if not has_printed and delta > CONNECT_TIMEOUT / 2:
+            print("...")
+            has_printed = True
+        elif delta > CONNECT_TIMEOUT:
+            break
+
+    if not has_printed:
+        print("...")  # keep the output consistent
+
+    return test_func()
+
+
+# AP
+def instance0():
+    ap = network.WLAN(network.WLAN.IF_AP)
+    ssid = "MP-test-" + os.urandom(6).hex()
+    psk = "Secret-" + os.urandom(6).hex()
+
+    # stop any previous activity
+    network.WLAN(network.WLAN.IF_STA).active(False)
+    ap.active(False)
+
+    ap.active(True)
+    ap.config(ssid=ssid, key=psk, channel=CHANNEL, security=network.WLAN.SEC_WPA_WPA2)
+
+    # print("AP setup", ssid, psk)
+    print("AP started")
+
+    multitest.globals(SSID=ssid, PSK=psk)
+    multitest.next()
+
+    # Wait for station
+    if not wait_for(ap.isconnected):
+        raise RuntimeError("Timed out waiting for station, status ", ap.status())
+
+    print("AP got station")
+    time.sleep(
+        3
+    )  # depending on port, may still need to negotiate DHCP lease for STA to see connection
+
```

Comment:
It might be worth testing the result of `ap.status('stations')` here... although that could open a can of inconsistent worms.

Reply:
Let's add that in a separate test.